coala_quickstart/generation/SettingsClass.py

```
@@ -0,0 +1,256 @@
+    """
+
+    def __init__(self, settings, function, function_name, bear, trigger):
+        """
+        :param settings:
+            Either a dict of non-optional settings of the form:
+            {'setting_name': ('Description.', <class 'type'>),}
+            or
+            a dict of optional settings of the form:
+            {'setting_name': default_values,}
+        :param function:
+            The function object i.e. either the run() method or
+            the create_arguments() method of the bear.
+        :param function_name:
+            Name of the function, either 'run' or 'create_arguments'
+        :param bear:
+            The current bear object.
+        :param trigger:
+            String of value either 'optional' or 'non-optional'
+            depending on type of settings
+        """
+        self.settings_bool = []
+        self.settings_others = []
+        self.fillup_settings(function, settings, bear, trigger)
+
+    def fillup_settings(self, function, settings, bear, trigger):
+        """
+        Fill settings_bool and settings_others depending upon whether the
+        particular setting by the bear takes a bool value or any other.
+        :param function:
+            The function object i.e. either the run() method or
+            the create_arguments() method of the bear.
+        :param settings:
+            Either a dict of non-optional settings of the form:
+            {'setting_name': ('Description.', <class 'type'>),}
+            or
+            a dict of optional settings of the form:
+            {'setting_name': default_values,}
+        :param bear:
+            The current bear object.
+        :param trigger:
+            String of value either 'optional' or 'non-optional'
+            depending on type of settings
+        """
+        for key in settings:
+            if trigger == 'optional':
+                self.fillup_optional_settings(key, function)
+            else:
```

this should be `elif trigger == 'non-optional'` and the else clause after that should raise a Value error with a proper message of Invalid trigger type.